commit 9d94d8020514954fc0304d3ac6d2a4e56283232c
Author: Joe Walker <jwalker@mozilla.com>
Date:   Wed Nov 10 14:20:25 2010 +0000

    convert layers to commonjs export format

diff --git a/plugins/ace/lib/ace/layer/cursor.js b/plugins/ace/lib/ace/layer/cursor.js
index 0440248..977ed49 100644
--- a/plugins/ace/lib/ace/layer/cursor.js
+++ b/plugins/ace/lib/ace/layer/cursor.js
@@ -145,5 +145,6 @@ var Cursor = function(parentEl) {
 
 }).call(Cursor.prototype);
 
-return Cursor;
+exports.Cursor = Cursor;
+
 });
diff --git a/plugins/ace/lib/ace/layer/gutter.js b/plugins/ace/lib/ace/layer/gutter.js
index 87db163..4202772 100644
--- a/plugins/ace/lib/ace/layer/gutter.js
+++ b/plugins/ace/lib/ace/layer/gutter.js
@@ -81,5 +81,6 @@ var Gutter = function(parentEl) {
 
 }).call(Gutter.prototype);
 
-return Gutter;
+exports.Gutter = Gutter;
+
 });
diff --git a/plugins/ace/lib/ace/layer/marker.js b/plugins/ace/lib/ace/layer/marker.js
index 0466f4b..1e571c8 100644
--- a/plugins/ace/lib/ace/layer/marker.js
+++ b/plugins/ace/lib/ace/layer/marker.js
@@ -181,5 +181,6 @@ var Marker = function(parentEl) {
 
 }).call(Marker.prototype);
 
-return Marker;
+exports.Marker = Marker;
+
 });
diff --git a/plugins/ace/lib/ace/layer/text.js b/plugins/ace/lib/ace/layer/text.js
index 57723b5..149fac5 100644
--- a/plugins/ace/lib/ace/layer/text.js
+++ b/plugins/ace/lib/ace/layer/text.js
@@ -297,5 +297,6 @@ var Text = function(parentEl) {
 
 }).call(Text.prototype);
 
-return Text;
+exports.Text = Text;
+
 });
diff --git a/plugins/ace/lib/ace/virtual_renderer.js b/plugins/ace/lib/ace/virtual_renderer.js
index 75a7a5e..e8facea 100644
--- a/plugins/ace/lib/ace/virtual_renderer.js
+++ b/plugins/ace/lib/ace/virtual_renderer.js
@@ -41,10 +41,10 @@ var oop = require("ace/lib/oop");
 var lang = require("ace/lib/lang");
 var dom = require("ace/lib/dom");
 var event = require("ace/lib/event");
-var GutterLayer = require("ace/layer/gutter");
-var MarkerLayer = require("ace/layer/marker");
-var TextLayer = require("ace/layer/text");
-var CursorLayer = require("ace/layer/cursor");
+var GutterLayer = require("ace/layer/gutter").Gutter;
+var MarkerLayer = require("ace/layer/marker").Marker;
+var TextLayer = require("ace/layer/text").Text;
+var CursorLayer = require("ace/layer/cursor").Cursor;
 var ScrollBar = require("ace/scrollbar");
 var RenderLoop = require("ace/renderloop");
 var MEventEmitter = require("ace/event_emitter").MEventEmitter;
